File: .sdk-repos/orchestration-cluster-api-rust/examples/backup.rs
//! Compilable usage examples for runtime backup operations.
//!
//! Region-tagged snippets are surfaced in the Camunda REST API documentation via
//! `examples/operation-map.json`, and are type-checked by `cargo build --examples`.
#![allow(dead_code, unused_variables, unused_imports)]

use camunda_orchestration_sdk::apis::backup_api::DeleteHistoryBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::DeleteHistoryBackupParams;
use camunda_orchestration_sdk::apis::backup_api::DeleteRuntimeBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::DeleteRuntimeBackupParams;
use camunda_orchestration_sdk::apis::backup_api::DeleteRuntimeBackupStateAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::GetHistoryBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::GetHistoryBackupParams;
use camunda_orchestration_sdk::apis::backup_api::GetRuntimeBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::GetRuntimeBackupParams;
use camunda_orchestration_sdk::apis::backup_api::GetRuntimeBackupStateAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::ListHistoryBackupsAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::ListHistoryBackupsParams;
use camunda_orchestration_sdk::apis::backup_api::ListRuntimeBackupsAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::ListRuntimeBackupsParams;
use camunda_orchestration_sdk::apis::backup_api::SyncRuntimeBackupStateAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::TakeHistoryBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::TakeHistoryBackupParams;
use camunda_orchestration_sdk::apis::backup_api::TakeRuntimeBackupAsClusterAdminParams;
use camunda_orchestration_sdk::apis::backup_api::TakeRuntimeBackupParams;
use camunda_orchestration_sdk::models::TakeHistoryBackupRequest;
use camunda_orchestration_sdk::models::TakeRuntimeBackupRequest;
use camunda_orchestration_sdk::CamundaClient;

// region TakeRuntimeBackup
async fn take_runtime_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .take_runtime_backup(TakeRuntimeBackupParams {
            take_runtime_backup_request: Some(TakeRuntimeBackupRequest {
                backup_id: Some(Some(1)),
            }),
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion TakeRuntimeBackup

// region ListRuntimeBackups
async fn list_runtime_backups() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let backups = client
        .list_runtime_backups(ListRuntimeBackupsParams { prefix: None })
        .await?;
    for backup in backups {
        println!("{backup:#?}");
    }

    Ok(())
}
// endregion ListRuntimeBackups

// region GetRuntimeBackup
async fn get_runtime_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_runtime_backup(GetRuntimeBackupParams { backup_id: 1 })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion GetRuntimeBackup

// region DeleteRuntimeBackup
async fn delete_runtime_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_runtime_backup(DeleteRuntimeBackupParams { backup_id: 1 })
        .await?;
    println!("Delete runtime backup: done");

    Ok(())
}
// endregion DeleteRuntimeBackup

// region GetRuntimeBackupState
async fn get_runtime_backup_state() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let state = client.get_runtime_backup_state().await?;
    println!("{state:#?}");

    Ok(())
}
// endregion GetRuntimeBackupState

// region SyncRuntimeBackupState
async fn sync_runtime_backup_state() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let state = client.sync_runtime_backup_state().await?;
    println!("{state:#?}");

    Ok(())
}
// endregion SyncRuntimeBackupState

// region DeleteRuntimeBackupState
async fn delete_runtime_backup_state() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client.delete_runtime_backup_state().await?;
    println!("Delete runtime backup state: done");

    Ok(())
}
// endregion DeleteRuntimeBackupState

// region TakeHistoryBackup
async fn take_history_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .take_history_backup(TakeHistoryBackupParams {
            take_history_backup_request: TakeHistoryBackupRequest { backup_id: 1 },
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion TakeHistoryBackup

// region ListHistoryBackups
async fn list_history_backups() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    // `prefix` must end in a single '*'. Setting `verbose` to false makes the
    // query cheaper, at the cost of snapshot-level detail.
    let backups = client
        .list_history_backups(ListHistoryBackupsParams {
            prefix: None,
            verbose: None,
        })
        .await?;
    for backup in backups {
        println!("{backup:#?}");
    }

    Ok(())
}
// endregion ListHistoryBackups

// region GetHistoryBackup
async fn get_history_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_history_backup(GetHistoryBackupParams { backup_id: 1 })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion GetHistoryBackup

// region DeleteHistoryBackup
async fn delete_history_backup() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_history_backup(DeleteHistoryBackupParams { backup_id: 1 })
        .await?;
    println!("Delete history backup: done");

    Ok(())
}
// endregion DeleteHistoryBackup

// region TakeRuntimeBackupAsClusterAdmin
async fn take_runtime_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    // Setting `physical_tenant_id` scopes the backup to one physical tenant; omit
    // it to back up every physical tenant of the cluster.
    let result = client
        .take_runtime_backup_as_cluster_admin(TakeRuntimeBackupAsClusterAdminParams {
            physical_tenant_id: Some("default".to_string()),
            take_runtime_backup_request: Some(TakeRuntimeBackupRequest {
                backup_id: Some(Some(1)),
            }),
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion TakeRuntimeBackupAsClusterAdmin

// region ListRuntimeBackupsAsClusterAdmin
async fn list_runtime_backups_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let backups = client
        .list_runtime_backups_as_cluster_admin(ListRuntimeBackupsAsClusterAdminParams {
            physical_tenant_id: None,
            prefix: None,
        })
        .await?;
    for backup in backups {
        println!("{backup:#?}");
    }

    Ok(())
}
// endregion ListRuntimeBackupsAsClusterAdmin

// region GetRuntimeBackupAsClusterAdmin
async fn get_runtime_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_runtime_backup_as_cluster_admin(GetRuntimeBackupAsClusterAdminParams {
            backup_id: 1,
            physical_tenant_id: None,
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion GetRuntimeBackupAsClusterAdmin

// region DeleteRuntimeBackupAsClusterAdmin
async fn delete_runtime_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_runtime_backup_as_cluster_admin(DeleteRuntimeBackupAsClusterAdminParams {
            backup_id: 1,
            physical_tenant_id: None,
        })
        .await?;
    println!("Delete runtime backup as cluster admin: done");

    Ok(())
}
// endregion DeleteRuntimeBackupAsClusterAdmin

// region GetRuntimeBackupStateAsClusterAdmin
async fn get_runtime_backup_state_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let state = client
        .get_runtime_backup_state_as_cluster_admin(GetRuntimeBackupStateAsClusterAdminParams {
            physical_tenant_id: None,
        })
        .await?;
    println!("{state:#?}");

    Ok(())
}
// endregion GetRuntimeBackupStateAsClusterAdmin

// region SyncRuntimeBackupStateAsClusterAdmin
async fn sync_runtime_backup_state_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let state = client
        .sync_runtime_backup_state_as_cluster_admin(SyncRuntimeBackupStateAsClusterAdminParams {
            physical_tenant_id: None,
        })
        .await?;
    println!("{state:#?}");

    Ok(())
}
// endregion SyncRuntimeBackupStateAsClusterAdmin

// region DeleteRuntimeBackupStateAsClusterAdmin
async fn delete_runtime_backup_state_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_runtime_backup_state_as_cluster_admin(
            DeleteRuntimeBackupStateAsClusterAdminParams {
                physical_tenant_id: None,
            },
        )
        .await?;
    println!("Delete runtime backup state as cluster admin: done");

    Ok(())
}
// endregion DeleteRuntimeBackupStateAsClusterAdmin

// region TakeHistoryBackupAsClusterAdmin
async fn take_history_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .take_history_backup_as_cluster_admin(TakeHistoryBackupAsClusterAdminParams {
            take_history_backup_request: TakeHistoryBackupRequest { backup_id: 1 },
            physical_tenant_id: None,
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion TakeHistoryBackupAsClusterAdmin

// region ListHistoryBackupsAsClusterAdmin
async fn list_history_backups_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    // `prefix` must end in a single '*'. Setting `verbose` to false makes the
    // query cheaper, at the cost of snapshot-level detail.
    let backups = client
        .list_history_backups_as_cluster_admin(ListHistoryBackupsAsClusterAdminParams {
            physical_tenant_id: None,
            prefix: None,
            verbose: None,
        })
        .await?;
    for backup in backups {
        println!("{backup:#?}");
    }

    Ok(())
}
// endregion ListHistoryBackupsAsClusterAdmin

// region GetHistoryBackupAsClusterAdmin
async fn get_history_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_history_backup_as_cluster_admin(GetHistoryBackupAsClusterAdminParams {
            backup_id: 1,
            physical_tenant_id: None,
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion GetHistoryBackupAsClusterAdmin

// region DeleteHistoryBackupAsClusterAdmin
async fn delete_history_backup_as_cluster_admin() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_history_backup_as_cluster_admin(DeleteHistoryBackupAsClusterAdminParams {
            backup_id: 1,
            physical_tenant_id: None,
        })
        .await?;
    println!("Delete history backup as cluster admin: done");

    Ok(())
}
// endregion DeleteHistoryBackupAsClusterAdmin

fn main() {
    // Examples above are compiled, not executed.
}

File: .sdk-repos/orchestration-cluster-api-php/examples/backup.php
<?php

/**
 * Compilable usage examples for runtime and history backup operations.
 *
 * Backup deletion, restoration, and cluster-admin methods change durable cluster state.
 * Run them only against the intended cluster after reviewing the requested backup ID.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region TakeRuntimeBackup
/**
 * Take a runtime backup.
 */
function take_runtime_backup(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\TakeRuntimeBackupRequest $takeRuntimeBackupRequest = null): void
{
    $client->takeRuntimeBackup($takeRuntimeBackupRequest);
}
// endregion TakeRuntimeBackup

// region ListRuntimeBackups
/**
 * List runtime backups.
 */
function list_runtime_backups(CamundaClient $client, ?string $prefix = null): void
{
    $client->listRuntimeBackups($prefix);
}
// endregion ListRuntimeBackups

// region GetRuntimeBackup
/**
 * Get a runtime backup.
 */
function get_runtime_backup(CamundaClient $client, int $backupId): void
{
    $client->getRuntimeBackup($backupId);
}
// endregion GetRuntimeBackup

// region DeleteRuntimeBackup
/**
 * Delete a runtime backup.
 */
function delete_runtime_backup(CamundaClient $client, int $backupId): void
{
    $client->deleteRuntimeBackup($backupId);
}
// endregion DeleteRuntimeBackup

// region GetRuntimeBackupState
/**
 * Get runtime backup state.
 */
function get_runtime_backup_state(CamundaClient $client): void
{
    $client->getRuntimeBackupState();
}
// endregion GetRuntimeBackupState

// region SyncRuntimeBackupState
/**
 * Sync runtime backup state.
 */
function sync_runtime_backup_state(CamundaClient $client): void
{
    $client->syncRuntimeBackupState();
}
// endregion SyncRuntimeBackupState

// region DeleteRuntimeBackupState
/**
 * Delete runtime backup state.
 */
function delete_runtime_backup_state(CamundaClient $client): void
{
    $client->deleteRuntimeBackupState();
}
// endregion DeleteRuntimeBackupState

// region TakeHistoryBackup
/**
 * Take a history backup.
 */
function take_history_backup(CamundaClient $client, \Camunda\Orchestration\Api\Model\TakeHistoryBackupRequest $takeHistoryBackupRequest): void
{
    $client->takeHistoryBackup($takeHistoryBackupRequest);
}
// endregion TakeHistoryBackup

// region ListHistoryBackups
/**
 * List history backups.
 */
function list_history_backups(CamundaClient $client, ?string $prefix = null, ?bool $verbose = true): void
{
    $client->listHistoryBackups($prefix, $verbose);
}
// endregion ListHistoryBackups

// region GetHistoryBackup
/**
 * Get a history backup.
 */
function get_history_backup(CamundaClient $client, int $backupId): void
{
    $client->getHistoryBackup($backupId);
}
// endregion GetHistoryBackup

// region DeleteHistoryBackup
/**
 * Delete a history backup.
 */
function delete_history_backup(CamundaClient $client, int $backupId): void
{
    $client->deleteHistoryBackup($backupId);
}
// endregion DeleteHistoryBackup

// region TakeHistoryBackupAsClusterAdmin
/**
 * Take a cluster history backup.
 * @param array<string, mixed> $variables
 */
function take_history_backup_as_cluster_admin(CamundaClient $client, \Camunda\Orchestration\Api\Model\TakeHistoryBackupRequest $takeHistoryBackupRequest, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->takeHistoryBackupAsClusterAdmin($takeHistoryBackupRequest, $physicalTenantId, $hostIndex, $variables);
}
// endregion TakeHistoryBackupAsClusterAdmin

// region ListHistoryBackupsAsClusterAdmin
/**
 * List cluster history backups.
 * @param array<string, mixed> $variables
 */
function list_history_backups_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?string $prefix = null, ?bool $verbose = true, ?int $hostIndex = null, array $variables = []): void
{
    $client->listHistoryBackupsAsClusterAdmin($physicalTenantId, $prefix, $verbose, $hostIndex, $variables);
}
// endregion ListHistoryBackupsAsClusterAdmin

// region GetHistoryBackupAsClusterAdmin
/**
 * Get a cluster history backup.
 * @param array<string, mixed> $variables
 */
function get_history_backup_as_cluster_admin(CamundaClient $client, int $backupId, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->getHistoryBackupAsClusterAdmin($backupId, $physicalTenantId, $hostIndex, $variables);
}
// endregion GetHistoryBackupAsClusterAdmin

// region DeleteHistoryBackupAsClusterAdmin
/**
 * Delete a cluster history backup.
 * @param array<string, mixed> $variables
 */
function delete_history_backup_as_cluster_admin(CamundaClient $client, int $backupId, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->deleteHistoryBackupAsClusterAdmin($backupId, $physicalTenantId, $hostIndex, $variables);
}
// endregion DeleteHistoryBackupAsClusterAdmin

// region TakeRuntimeBackupAsClusterAdmin
/**
 * Take a cluster runtime backup.
 * @param array<string, mixed> $variables
 */
function take_runtime_backup_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?\Camunda\Orchestration\Api\Model\TakeRuntimeBackupRequest $takeRuntimeBackupRequest = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->takeRuntimeBackupAsClusterAdmin($physicalTenantId, $takeRuntimeBackupRequest, $hostIndex, $variables);
}
// endregion TakeRuntimeBackupAsClusterAdmin

// region ListRuntimeBackupsAsClusterAdmin
/**
 * List cluster runtime backups.
 * @param array<string, mixed> $variables
 */
function list_runtime_backups_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?string $prefix = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->listRuntimeBackupsAsClusterAdmin($physicalTenantId, $prefix, $hostIndex, $variables);
}
// endregion ListRuntimeBackupsAsClusterAdmin

// region GetRuntimeBackupStateAsClusterAdmin
/**
 * Get cluster runtime backup state.
 * @param array<string, mixed> $variables
 */
function get_runtime_backup_state_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->getRuntimeBackupStateAsClusterAdmin($physicalTenantId, $hostIndex, $variables);
}
// endregion GetRuntimeBackupStateAsClusterAdmin

// region DeleteRuntimeBackupStateAsClusterAdmin
/**
 * Delete cluster runtime backup state.
 * @param array<string, mixed> $variables
 */
function delete_runtime_backup_state_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->deleteRuntimeBackupStateAsClusterAdmin($physicalTenantId, $hostIndex, $variables);
}
// endregion DeleteRuntimeBackupStateAsClusterAdmin

// region SyncRuntimeBackupStateAsClusterAdmin
/**
 * Sync cluster runtime backup state.
 * @param array<string, mixed> $variables
 */
function sync_runtime_backup_state_as_cluster_admin(CamundaClient $client, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->syncRuntimeBackupStateAsClusterAdmin($physicalTenantId, $hostIndex, $variables);
}
// endregion SyncRuntimeBackupStateAsClusterAdmin

// region GetRuntimeBackupAsClusterAdmin
/**
 * Get a cluster runtime backup.
 * @param array<string, mixed> $variables
 */
function get_runtime_backup_as_cluster_admin(CamundaClient $client, int $backupId, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->getRuntimeBackupAsClusterAdmin($backupId, $physicalTenantId, $hostIndex, $variables);
}
// endregion GetRuntimeBackupAsClusterAdmin

// region DeleteRuntimeBackupAsClusterAdmin
/**
 * Delete a cluster runtime backup.
 * @param array<string, mixed> $variables
 */
function delete_runtime_backup_as_cluster_admin(CamundaClient $client, int $backupId, ?string $physicalTenantId = null, ?int $hostIndex = null, array $variables = []): void
{
    $client->deleteRuntimeBackupAsClusterAdmin($backupId, $physicalTenantId, $hostIndex, $variables);
}
// endregion DeleteRuntimeBackupAsClusterAdmin
